### PR TITLE
Add Binary Tree traversal algorithms. Update README.

### PR DESCRIPTION
**Pull request by alpha74**
Add tree algorithms.
- Inorder traversal of binary tree.
- Preorder traversal of binary tree.
- Postorder traversal of binary tree

### CODE REVIEW

**Comment by alpha74** [comment]
For Issue #111 @Bhavnaharitsa

**Comment by alpha74** [comment]
Update:
 - Add height of binary tree.

**Comment by alpha74** [comment]
Add links to main README.

**Comment by alpha74** [comment]
Sort titles in main README. Issue #114

**Comment by alpha74** [comment]
Fix Issue #115

File: Algorithms/Trees/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Tree Algorithms
+
+## List
```
Comment (Bhavnaharitsa) [inline Algorithms/Trees/README.md:3]: no need this

File: Algorithms/Trees/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Tree Algorithms
```
Comment (Bhavnaharitsa) [inline Algorithms/Trees/README.md:1]: add ##

File: Algorithms/Trees/inorder_traversal_bt.cpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
+// Inorder traversal of a binary tree
+// Implemented in C++
+
+/*
+	Function recursively calls the left and right children and prints the data of current node/
+*/
+
```
Comment (Bhavnaharitsa) [inline Algorithms/Trees/inorder_traversal_bt.cpp:21]: push all comments in single /*….*/

File: Algorithms/Trees/inorder_traversal_bt.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		right = NULL ;  
+	} 
+} ; 
+ 
+
+// Print Inorder traversal
+int printInorder(struct Node* node) 
+{ 
+	if (node == NULL) 
+		return;
```
Comment (Bhavnaharitsa) [inline Algorithms/Trees/inorder_traversal_bt.cpp]: return statement with no value? Will show error

File: Algorithms/Trees/inorder_traversal_bt.cpp [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+} 
+ 
+// Main program
+int main() 
+{ 
+	// Create a tree manually.
+	struct Node *root = new Node(2) ; 
+
+	root -> left = new Node( 3 ) ; 
+	root -> right = new Node( 4 ) ;
```
Comment (Bhavnaharitsa) [inline Algorithms/Trees/inorder_traversal_bt.cpp:68]: make it dynamic

File: Algorithms/Trees/inorder_traversal_bt.cpp [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+// Declaration of data strucutre of binary tree.
+struct Node 
+{ 
+	int data ; 
+	struct Node* left, *right ; 
+
+	Node(int d ) 
+	{ 
+		data = d ;
```
Comment (Bhavnaharitsa) [inline Algorithms/Trees/inorder_traversal_bt.cpp:38]: what is d?

**Comment by Bhavnaharitsa** [comment]
Umm.. a better approach could be used. This consumes lot of time complexity

**Comment by Bhavnaharitsa** [comment]
> For Issue #111 @Bhavnaharitsa

new issue should have new pr

**Comment by Bhavnaharitsa** [comment]
Umm...this pr solves 3 issues which violates the rules. A pr should solve only one issue. 
Closing this

**Comment by alpha74** [comment]
> 
> 
> Umm.. a better approach could be used. This consumes lot of time complexity

Hi, which program are you talking about here? @Bhavnaharitsa

**Comment by Bhavnaharitsa** [comment]
Every program...inorder, preorder and postorder

**Comment by akashksinghal** [comment]
Should I try the Tree Algorithms and add those ?

**Pull request closed without merging**